cmd/compare.go

```diff
@@ -254,3 +322,22 @@ func validateTypes(old *schema.TypeSpec, new *schema.TypeSpec, prefix string) (v
 func formatName(provider, s string) string {
 	return strings.ReplaceAll(strings.TrimPrefix(s, fmt.Sprintf("%s:", provider)), ":", ".")
 }
+
+type set[T comparable] struct{ m map[T]struct{} }
+
+func newSetFromList[T comparable](elems []T) set[T] {
```

Comment:
apologies for not saying this earlier - what is the "list" we are creating a set from? if there's no more descriptive name, (`Elements`? `ElementList`?) then a comment would be dope

Reply:
The list is `elems`. It is an arbitrary set from an arbitrary slice. I'll change the name to `Slice` since I believe it is more idiomatic go.